src/parser.rs

```diff
@@ -423,6 +423,7 @@ impl<'a> Parser<'a> {
                 Keyword::TRY_CAST => self.parse_try_cast_expr(),
                 Keyword::EXISTS => self.parse_exists_expr(),
                 Keyword::EXTRACT => self.parse_extract_expr(),
+                Keyword::POSITION => self.parse_position_expr(),
```

Comment:
@alamb This section is not a must. If I will remove it, the parsed object will look like:

```
"projection": [
            {
              "UnnamedExpr": {
                "Function": {
                  "name": [
                    {
                      "value": "POSITION",
                      "quote_style": null
                    }
                  ],
                  "args": [
                    {
                      "Unnamed": {
                        "Expr": {
                          "Position": {
                            "expr": {
                              "Value": {
                                "SingleQuotedString": "@"
                              }
                            },
                            "from": {
                              "Identifier": {
                                "value": "field",
                                "quote_style": null
                              }
                            }
                          }
                        }
                      }
                    }
                  ],
                  "over": null,
                  "distinct": false
                }
              }
            }
```

I'm not if position deserves special method in the parser.

Reply:
> I'm not if position deserves special method in the parser.

I guess if there is some reason to special case the parser for `POSITION` then it is a good idea. If the parser can handle `POSITION` without any changes, I would be inclined not to include the extra code (and any users of `sqlparser-rs` who want to support `POSITION` can look for the patterns they want to support)